### PR TITLE
feat: add settings for files list views

### LINKED ISSUE #1516: Display mode of diff view should be remembered as setting when restarting Vorta
I use the latest Flatpak version 0.8.9.

### PR DESCRIPTION
**Pull request by diivi**
Added two settings to remember user's preferred files list view in the diff and extract dialogs
- `extract_files_display_mode`
- `diff_files_display_mode`

fixes #1516 
### Motivation and Context
Improves UX

### How Has This Been Tested?
Tested by changing the views and reopening vorta.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://vorta.borgbase.com/contributing/) guide.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

I have to update the tests, then I'll make this ready for review.

*I provide my contribution under the terms of the [license](./../../LICENSE.txt) of this repository and I affirm the [Developer Certificate of Origin][dco].*

[dco]: https://developercertificate.org/

### CODE REVIEW

File: src/vorta/views/partials/treemodel.py [hunk 1]
```diff
@@ -323,14 +323,12 @@ class DisplayMode(enum.Enum):
         #: simple list of items
         FLAT = enum.auto()
 
-    def __init__(self, parent=None):
+    def __init__(self, mode, parent=None):
```
Comment (real-yfprojects) [inline src/vorta/views/partials/treemodel.py:326]:
```suggestion
    def __init__(self, mode: 'FileTreeModel.DisplayMode', *, parent=None):
```

Comment (diivi) [reply src/vorta/views/partials/treemodel.py]: What was the star for here? I added it and got errors in tests, so I kept the type annotation suggestion

Comment (real-yfprojects) [reply src/vorta/views/partials/treemodel.py]: It forces to use `parent` as a kw arg. I like that for QObjects since it will increase forward compatibility.

File: src/vorta/views/archive_tab.py [hunk 2]
```diff
@@ -714,12 +715,24 @@ def extract_action(self):
         else:
             self._set_status(self.tr('Select an archive to restore first.'))
 
+    def get_extract_display_mode(self):
```
Comment (real-yfprojects) [inline src/vorta/views/archive_tab.py:718]: The name isn't fully fitting. The method doesn't return the display mode but instantiates a TreeModel.

File: src/vorta/views/diff_result.py [hunk 3]
```diff
@@ -97,6 +98,8 @@ def __init__(self, archive_newer, archive_older, model: 'DiffTree'):
         self.archiveNameLabel_1.setText(f'{archive_newer.name}')
         self.archiveNameLabel_2.setText(f'{archive_older.name}')
 
+        diff_result_display_mode = SettingsModel.get(key='diff_files_display_mode').str_value
+        self.comboBoxDisplayMode.setCurrentIndex(int(diff_result_display_mode))
         self.comboBoxDisplayMode.currentIndexChanged.connect(self.change_display_mode)
```
Comment (real-yfprojects) [inline src/vorta/views/diff_result.py:103]: If you swap these lines, do you still need `get_diff_tree`?

File: src/vorta/views/diff_result.py [hunk 4]
```diff
@@ -845,3 +852,6 @@ def data(self, index: QModelIndex, role: int = Qt.ItemDataRole.DisplayRole):
                 )
 
             return tooltip
+
+    def __init__(self, mode=None, parent=None):
+        super().__init__(mode, parent)
```
Comment (real-yfprojects) [inline src/vorta/views/diff_result.py:857]: Why is that needed?

Comment (diivi) [reply src/vorta/views/diff_result.py:857]: Yeah, with mode=None this might not be needed. Earlier when I was passing a mode i used this. Thanks for pointing this out.

File: src/vorta/views/diff_result.py (hunk 3 above)
Comment (diivi) [reply src/vorta/views/diff_result.py:103]: Didn't get you 😕 Could you please elaborate?

Comment (real-yfprojects) [reply src/vorta/views/diff_result.py:103]: lines 102 and 103. The signal will triggered by the programmatic change too.

Comment (diivi) [reply src/vorta/views/diff_result.py:103]: Ah, nice!

File: src/vorta/views/diff_result.py [hunk 5]
```diff
@@ -97,7 +98,9 @@ def __init__(self, archive_newer, archive_older, model: 'DiffTree'):
         self.archiveNameLabel_1.setText(f'{archive_newer.name}')
         self.archiveNameLabel_2.setText(f'{archive_older.name}')
 
+        diff_result_display_mode = SettingsModel.get(key='diff_files_display_mode').str_value
         self.comboBoxDisplayMode.currentIndexChanged.connect(self.change_display_mode)
+        self.comboBoxDisplayMode.setCurrentIndex(int(diff_result_display_mode))
```
Comment (real-yfprojects) [inline src/vorta/views/diff_result.py:103]:
```suggestion
        self.comboBoxDisplayMode.currentIndexChanged.connect(self.change_display_mode)
        self.comboBoxDisplayMode.setCurrentIndex(int(SettingsModel.get(key='diff_files_display_mode').str_value))
```

This or move line 101 down one line.

**Review by real-yfprojects — DISMISSED** [review]
Works as expected! Thank you.

**Review by m3nu — APPROVED** [review]
